controlpanel/frontend/views/app.py

```diff
@@ -362,6 +373,46 @@ class EnableTextractApp(PolicyAccessMixin, UpdateView):
     method_name = "set_textract_access"
 
 
+class UpdateCloudPlatformRoleArn(
+    OIDCLoginRequiredMixin,
+    PermissionRequiredMixin,
+    UpdateView,
+):
+    form_class = CloudPlatformArnForm
+    model = App
+    permission_required = "api.add_cloud_platform_arn"
+
+    def get_form_kwargs(self):
+        kwargs = FormMixin.get_form_kwargs(self)
+        return kwargs
+
+    def get_success_url(self):
+        return reverse_lazy("manage-app", kwargs={"pk": self.object.id})
+
+    def form_valid(self, form):
+        try:
+            with transaction.atomic():
+                app = self.get_object()
+                app.cloud_platform_role_arn = form.cleaned_data.get("cloud_platform_role_arn")
+                app.save()
+                app.update_cloud_platform_access()
+                messages.success(self.request, "Successfully updated cloud platform ARN")
+                return HttpResponseRedirect(self.get_success_url())
+
+        except Exception as e:
```

Comment:
I know that catching `Exception` is not uncommon already in the codebase, but its not great practice. If you want to catch an exception to handle errors on the frontend more gracefully than just letting the 500 page raise, I would suggest two options:

1. Catch `botocore.exceptions.ClientError` here, log it to sentry, then proceed to redirect the user and display your 'friendly' error message.
2. Catch `botocore.exceptions.ClientError` in `aws.py` (specifically `MalformedPolicyDocumentException` perhaps?) and then raise a custom exception that we catch here so we can log it in sentry, then redirect the user with your 'friendly' error message